### PR TITLE
378_kth_smallest_element.py

### PR DESCRIPTION
**Pull request by Rwatana**
Please check my code

### CODE REVIEW

File: 378_kth_smallest_element.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from typing import List
+
+
+class Solution:
+    def kthSmallest(self, matrix: List[List[int]], k: int) -> int:
```
Comment (ryujumoto) [inline 378_kth_smallest_element.py:11]: You can use this with more efficient solution.

**Review by ryujumoto — COMMENTED** [review]
Done.

**Comment by Rwatana** [comment]
@Ryuju0911 
```
class Solution:
    def kthSmallest(self, matrix: List[List[int]], k: int) -> int:
        num = []
        for i in range(len(matrix)):
            num += matrix[i]
        num.sort()
        return num[k-1]
```
I knew the upper one work well.
But, I couldn't understand the reason why the lower one got mistakes.
```
class Solution:
    def kthSmallest(self, matrix: List[List[int]], k: int) -> int:
        num = []
        for i in range(len(matrix)):
            num.append(matrix[i])
        num.sort()
        return num[k-1]
```
I investigated the rule of append. but, I couldn' t understand.
Please tell me.

**Comment by ryujumoto** [comment]
https://note.nkmk.me/python-list-append-extend-insert/

**Comment by ryujumoto** [comment]
Refer [this code](https://github.com/okuda-seminar/leetcode_answers/blob/main/378_kth-smallest-element-in-a-sorted-matrix.py).

**Comment by Rwatana** [comment]
I learned the way of using bisect. I fixed my code according to the reference.

**Review by ryujumoto — APPROVED** [review]

**Pull request merged**